shell/platform/darwin/common/framework/Source/FlutterNSBundleUtils.mm

```diff
@@ -54,15 +54,17 @@
 }
 
 NSString* FLTAssetPath(NSBundle* bundle) {
-  return [bundle objectForInfoDictionaryKey:@"FLTAssetsPath"] ?: kDefaultAssetPath;
+  return [bundle objectForInfoDictionaryKey:@"FLTAssetsPath"] ?: @"flutter_assets";
 }
 
-NSURL* FLTAssetsURLFromBundle(NSBundle* bundle) {
+NSString* FLTAssetsPathFromBundle(NSBundle* bundle) {
   NSString* flutterAssetsPath = FLTAssetPath(bundle);
-  NSURL* assets = [bundle URLForResource:flutterAssetsPath withExtension:nil];
+  // Use the raw path solution so that asset path can be returned from unloaded bundles.
+  // See https://github.com/flutter/engine/pull/46073
+  NSString* assetsPath = [bundle pathForResource:flutterAssetsPath ofType:@""];
```

Comment:
Optional nit: Is there a reason you're not using `nil` instead of `@""` here and below, given that they do the same thing (according to the docs)? `nil` is more common as "no value" than an empty string.

Reply:
I just reverted back to the original implementation, unsure why it was @"" instead of nil. I will change it to nil